docs/database-connection-string.md

```diff
@@ -1,85 +1,83 @@
+Value               | Description
+:------------------ | :---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+`RequireStarted`    | Expects the database process to already be running and will connect to it if it is, or throw an exception if it is not.
+`StartIfNotRunning` | Connects to the database process if it is already running, or starts and connects to it if it is not.
+`StartExclusive`    | Expects no database process to be already running, and starts and connects to a new process if that is the case. If a database process is already running, an exception is thrown.
 
-* `Open` - Opens the database if it exist.
-* `CreateIfNotExists` - Opens the database if it exist, and creates one if it doesn't.
+### StopMode
 
-### The `DatabaseStartMode` enumeration values
+The StopMode option defines whether to stop the database process once the connection to it is closed, for example when an application is terminated.
 
-* `RequireStarted` - Expects database processes to be running, and won't attempt starting them.
-* `StartIfNotRunning` - Connects if database processes are already running, and start them first if they are not.
-* `StartExclusive` - Expects no database processes to be running, and start them if they are not. Refuse to connect if database processes are already running.
+Value         | Description
+:------------ | :-----------------------------------------------------------------------------------
+`IfWeStarted` | Stops the database process if and only if it was started by the current application.
+`Never`       | Leaves the database process running, regardless of how it was started.
 
-### The `DatabaseStopMode` enumeration values
+**Note:** the `Never` option is not effective when the application process is forcibly killed, for example from a task manager or by `Ctrl + C`. In this case operating system will kill the corresponding database process as well.
 
-* `IfWeStarted` - Stops database processes if they where started by the current application.
-* `Never` - Leaves database processes running, even after the application shuts down and even if the application started them.
+### ContextCount
 
-_Note: the `Never` option is not effective when the application process is forcibly killed with task manager or `Ctrl + C`. In this case operating system will kill corresponding Starcounter processes as well._
+The ContextCount option defines the number of **database contexts** to use in the database connection.
 
-### The `ContextCount` option
+"Database context", as used here, should not be confused with `IDatabaseContext` as used in the .NET API for transactions. In the context of database connections, database contexts refer to the entrypoints to the database from the applications' point of view. When an application thread is assigned a database context, it can read and write to the database.
```

Comment:
When an application thread has a database context assigned, it can read and write to the database.

The same database context may be used between multiple applications threads, for example with the `async`/`await`. Starcounter manages this automatically.

Reply:
The automatic management you mention is implemented by a custom synchronization context, right? Or is that just for transactions? Or are they the same notion?

Reply:
It's the same thing, done through the custom synchronization context.